src/ctia/schemas/graphql/helpers.clj

```diff
@@ -372,17 +395,20 @@
    ^String description
    type-resolver-fn
    types]
-  (let [type-resolver (fn->type-resolver type-resolver-fn)
-        graphql-union (-> (GraphQLUnionType/newUnionType)
-                          (.description description)
-                          (.name union-name)
-                          ; FIXME: this method is deprecated
-                          (.typeResolver type-resolver))]
-    (doseq [type types]
-      (if (instance? GraphQLObjectType type)
-        (.possibleType graphql-union ^GraphQLObjectType type)
-        (.possibleType graphql-union ^GraphQLTypeReference type)))
-    (.build graphql-union)))
+    (get-or-update-type-registry
```

Comment:
Unions cannot be redefined, as I found out the hard way. As it stands, CTIA doesn't currently doesn't redefined any unions, so the registry wasn't needed.

However, the main strategy of porting to Trapperkeeper is for GraphQL constructors like this one (`new-union`) to return a function that accepts services. Hence, all consumers of a particular GraphQL union will attempt to create it, so the previous invariant is immediately broken for all unions with more than one occurrence in the code base.